plugin-gestures/src/test/java/com/mapbox/maps/plugin/gestures/GesturesPluginTest.kt

```diff
@@ -1017,8 +780,146 @@ class GesturePluginTest {
+  private val mapDelegateProvider: MapDelegateProvider = mockk(relaxUnitFun = true)
+  private val gesturesManager: AndroidGesturesManager = mockk(relaxed = true)
+  private val mapCameraManagerDelegate: MapCameraManagerDelegate = mockk(relaxUnitFun = true)
+  private val mapTransformDelegate: MapTransformDelegate = mockk(relaxUnitFun = true)
+  private val mapPluginProviderDelegate: MapPluginProviderDelegate = mockk(relaxUnitFun = true)
+  private val cameraAnimationsPlugin: CameraAnimationsPlugin = mockk(relaxed = true)
+
+  private lateinit var presenter: GesturesPluginImpl
+
+  private val typedArray: TypedArray = mockk(relaxed = true)
+  private val pack = "com.mapbox.maps"
+
+  @Before
+  fun prepare() {
+    mockkObject(GesturesAttributeParser)
+    every {
+      GesturesAttributeParser.parseGesturesSettings(
+        context,
+        attrs,
+        any()
+      )
+    } returns GesturesSettings()
+
+    every { context.obtainStyledAttributes(any(), any(), 0, 0) } returns typedArray
+    every { context.packageName } returns pack
+    every { typedArray.getString(any()) } returns "pk.token"
+    every { typedArray.getBoolean(any(), any()) } returns true
+    every { typedArray.getDimension(any(), any()) } returns 10.0f
+    every { typedArray.getFloat(any(), any()) } returns 10.0f
+    every { typedArray.getInt(any(), any()) } returns 2
+    every { typedArray.hasValue(any()) } returns true
+
+    every { mapDelegateProvider.mapCameraManagerDelegate } returns mapCameraManagerDelegate
+    every { mapDelegateProvider.mapTransformDelegate } returns mapTransformDelegate
+    every { mapDelegateProvider.mapPluginProviderDelegate } returns mapPluginProviderDelegate
+    every { mapPluginProviderDelegate.getPlugin<CameraAnimationsPlugin>(Plugin.MAPBOX_CAMERA_PLUGIN_ID) } returns cameraAnimationsPlugin
+
+    presenter = GesturesPluginImpl(context, attrs, mockk(relaxed = true))
+
+    presenter.bind(context, gesturesManager, attrs, 1f)
+    presenter.onDelegateProvider(mapDelegateProvider)
+    presenter.initialize()
+  }
+
+  @Test
+  fun testFling() {
+    every { mapCameraManagerDelegate.cameraState } returns CameraState(
+      Point.fromLngLat(0.0, 0.0),
+      EdgeInsets(0.0, 0.0, 0.0, 0.0),
+      0.0,
+      0.0,
+      targetPitch
+    )
+    every {
+      mapCameraManagerDelegate.getDragCameraOptions(
+        any(),
+        any()
+      )
+    } returns CameraOptions.Builder().build()
+    presenter.updateSettings { scrollMode = targetScrollMode }
+    val motionEvent = mockk<MotionEvent>()
+    every { motionEvent.x } returns 0.0f
+    every { motionEvent.y } returns 0.0f
+    val result = presenter.handleFlingEvent(motionEvent, mockk(), velocity.first, velocity.second)
+    verify {
+      mapCameraManagerDelegate.getDragCameraOptions(
+        ScreenCoordinate(0.0, 0.0),
+        expectedCoordinate
+      )
+    }
+    verify { cameraAnimationsPlugin.easeTo(any(), any()) }
```

Comment:
do you know exact values for this function? better not to use `any()`

Reply:
these values would depend on the mock value of 
```
getDragCameraOptions(
        ScreenCoordinate(0.0, 0.0),
        expectedCoordinate
      )
```

Reply:
added the mock value to easeTo function and also added check for animation duration.